### PR TITLE
feat(cell): Add a simple `Broadcast` implementation between `RegTile` and reduce tile.

### LINKED ISSUE #107: Broadcast the Reduce results into the `RegTile`.
The current shape of the reduce result on the `RegTile` is inconsistent with the shape of the `RegTile` itself. This has created difficulties in performing computations between these two different shapes (this would require a lot of redundant code to handle the shape differences). Therefore, I hope to keep the reduce result on the original `RegTile`, so that the `RegTile` can maintain a consistent shape to perform the computations.

I will give an example to illustrate the necessity of this form:

```cpp
// Execute reduce operation.
compute::MaxReduce<SrcReduceTile, kLayout> row_max;
// accumulate onto the max_vec
row_max(attn_block, max_vec);

__syncthreads();

// subtract max from flashattention -- now all <= 0.
compute::RegSubReduceTile<SrcReduceTile, kLayout> sub_max;
sub_max(attn_block, max_vec);
```

In this example, we first perform a reduce operation to compute the maximum value of each row, and then subtract the reduced maximum values from the `attn_block`. However, since the shapes of the two tiles (`attn_block` and the reduced result) are different, we have to implement different computation methods for the two different shapes, which will lead to additional code complexity.

Therefore, this will lead to different combination scenarios, and it will not be possible to implement it with just a single implementation for two identical shapes. Therefore, I feel that storing the Reduce result in the same format as the `BaseTile` might be a better choice. However, this could potentially increase the memory pressure.

I would very much like to get your advice. @haruhi55

### PR DESCRIPTION
**Pull request by KuangjuX**
Resolved #107 .

### CODE REVIEW

File: include/cell/compute/broadcast.hpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    using DType = typename DstTile::DType::DType;
+
+    static constexpr int kRows = DstTile::kRows;
+    static constexpr int kCols = DstTile::kCols;
+
+    DEVICE void operator()(const SrcTile& src, DstTile& dst) {}
+};
+
+template <typename SrcTile, typename DstTile>
+struct Broadcast<SrcTile, DstTile, tl::Layout::kRowMajor> {
```
Comment (haruhi55) [inline include/cell/compute/broadcast.hpp:21]: Does this implementation perform a broadcast assignment? Are broadcast operations like add, subtract, etc., necessary, and do you plan to support them?

Comment (KuangjuX) [reply include/cell/compute/broadcast.hpp:21]: Incorporating operations like Add and Sub into Broadcast also seems feasible. My original idea was to place them in Elementwise and Binary, and then execute the operations after the broadcasting?

Comment (KuangjuX) [reply include/cell/compute/broadcast.hpp:21]: It seems we could add a `Functor` template to do this. Taking an extra step doesn't seem to excessively introduce additional code complexity.

Comment (haruhi55) [reply include/cell/compute/broadcast.hpp:21]: Oh. I see.

**Review by haruhi55 — APPROVED** [review]
LGTM. Let's merge this PR.

**Pull request merged**